src/pages/home/report/ReportTypingIndicator.js

```diff
@@ -2,80 +2,75 @@ import PropTypes from 'prop-types';
+import useLocalize from '@hooks/useLocalize';
+import useNetwork from '@hooks/useNetwork';
+import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
+import * as ReportUtils from '@libs/ReportUtils';
 import useThemeStyles from '@styles/useThemeStyles';
-import * as PersonalDetails from '@userActions/PersonalDetails';
 import ONYXKEYS from '@src/ONYXKEYS';
 
 const propTypes = {
     /** Key-value pairs of user accountIDs/logins and whether or not they are typing. Keys are accountIDs or logins. */
     userTypingStatuses: PropTypes.objectOf(PropTypes.bool),
-
-    /** Information about the network */
-    network: networkPropTypes.isRequired,
-
-    ...withLocalizePropTypes,
 };
 
 const defaultProps = {
     userTypingStatuses: {},
 };
 
-function ReportTypingIndicator(props) {
+function ReportTypingIndicator({userTypingStatuses}) {
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
+
     const styles = useThemeStyles();
-    const usersTyping = useMemo(() => _.filter(_.keys(props.userTypingStatuses), (loginOrAccountID) => props.userTypingStatuses[loginOrAccountID]), [props.userTypingStatuses]);
+    const usersTyping = useMemo(() => _.filter(_.keys(userTypingStatuses), (loginOrAccountID) => userTypingStatuses[loginOrAccountID]), [userTypingStatuses]);
+    const firstUserTyping = usersTyping[0];
+    const isUserTypingADisplayName = Number.isNaN(Number(firstUserTyping));
+
+    const firstUserTypingID = useMemo(
+        () => (firstUserTyping && isUserTypingADisplayName ? PersonalDetailsUtils.getAccountIDsByLogins([firstUserTyping])[0] : firstUserTyping),
+        [firstUserTyping, isUserTypingADisplayName],
+    );
+
     // If we are offline, the user typing statuses are not up-to-date so do not show them
-    if (props.network.isOffline) {
+    if (isOffline || !firstUserTyping) {
         return null;
     }
 
-    const numUsersTyping = _.size(usersTyping);
-
-    // Decide on the Text element that will hold the display based on the number of users that are typing.
-    switch (numUsersTyping) {
-        case 0:
-            return null;
-
-        case 1:
-            return (
-                <TextWithEllipsis
-                    leadingText={PersonalDetails.getDisplayNameForTypingIndicator(usersTyping[0], props.translate('common.someone'))}
-                    trailingText={` ${props.translate('reportTypingIndicator.isTyping')}`}
-                    textStyle={[styles.chatItemComposeSecondaryRowSubText]}
-                    wrapperStyle={[styles.chatItemComposeSecondaryRow, styles.flex1]}
-                    leadingTextParentStyle={styles.chatItemComposeSecondaryRowOffset}
-                />
-            );
+    // If the user is typing on OldDot, firstUserTyping will be a string (the user's displayName)
+    const firstUserTypingDisplayName = isUserTypingADisplayName ? firstUserTyping : ReportUtils.getDisplayNameForParticipant(firstUserTypingID, false, false);
```

Comment:
If there's no case of `login` as `firstUserTyping`, this code can be updated like this:
```suggestion
    const firstUserTypingDisplayName = isUserTypingADisplayName ? firstUserTyping : ReportUtils.getDisplayNameForParticipant(firstUserTyping, false, false);
```
And remove unnecessary `firstUserTypingID`

Reply:
So is there a posibility that firstUserTyping is a login or not? In which case then? Maybe I'm missing something

Reply:
Do you agree with my code suggestion above?